jupyter_core/command.py

```diff
@@ -197,12 +216,13 @@ def main():
                     for p in path:
                         print('    ' + p)
             return
-    
+
     if not subcommand:
         parser.print_usage(file=sys.stderr)
         sys.exit("subcommand is required")
-    
-    command = 'jupyter-' + subcommand
+
+    command = _jupyter_abspath(sys.argv[0], subcommand)
```

Comment:
I think we always want to be looking up `jupyter-xyz`, regardless of what `sys.argv[0]` is.

Reply:
@takluyver I changed that considering the follow case reported by @mlucool 

```
Setup:
1. Set your PATH to have python 2 as your default version of python (prepend the path)
2. Run jupyterlab from a python 3 location (e.g. `path/to/python-3/bin/jupyterlab')
3. In a python 2 kernel run the following:
```
import sys
print(sys.version)
import os
print(os.environ['PATH'])
```
As expected, sys.version will be python 2. But now you can see that the PATH is modified with python3 in preprended. So if I were to do:
```
import subprocess
print(subprocess.Popen("python -c 'import sys; print(sys.version)'", shell=True, stdout=subprocess.PIPE).stdout.read())
```
This would show python 3.

This is really just a problem as we are migrating from python 2 to 3 and we want to run the server with 3 (as we want to use latest) but anything that is called via subprocess should use whatever you had in your PATH. We are planning to locally path `ipykernel_launcher.py` in a hacky way. This is suboptiomal so we'd like to revert this as soon as we can.
```

maybe it will not be a problem considering that there is no jupyterlab for py2 .. but if `path/to/python-3/bin/jupyterlab` is not in `$PATH` even if I run `jupyerlab` using the full path it will not be found.

do you have any thoughts here, considering this case?

Reply:
I'm fine with not modifying `$PATH`.

The `sys.argv[0]` that you pass in here is only used, as far as I can see, in this line:

```python
jupyter_subcommand = '{}-{}'.format(path, subcommand)
```

If `sys.argv[0]` is, e.g. `/home/takluyver/.local/bin/jupyter`, then you construct a subcommand like `/home/takluyver/.local/bin/jupyter-notebook`. That short-circuits `shutil.which()`,  so it will only work if that's the location of `jupyter-notebook`.

If the subcommand is `notebook`, then we should always be looking up `jupyter-notebook`, not `{sys.argv[0]}-notebook`.

Reply:
I see .. OK .. so maybe related to the scenario presented by @mlucool the workaround should be call `/path/jupyterlab-subcommand` directly

Reply:
@takluyver it is done for a new review! thanks!